content/vendor/guides/persistent-datastores.md

```diff
@@ -0,0 +1,940 @@
+          when: repl{{ ConfigOptionEquals "use_ingress" "1" }}
+```
+
+To add a database section, we'll modify it to include some database settings. In this case we'll remove the Ingress toggle that is included as an example, although you might also choose to leave this in. None of these database settings will have any effect yet, but we'll still be able to preview what the end user will see. Modify your yaml to include this database section:
+
+```yaml
+apiVersion: kots.io/v1beta1
+kind: Config
+metadata:
+  name: config-sample
+spec:
+  groups:
+    - name: database
+      title: Database
+      items:
+        - name: postgres_type
+          help_text: Would you like to use an embedded postgres instance, or connect to an external instance that you manage?
+          type: select_one
+          title: Postgres
+          default: embedded_postgres
+          items:
+            - name: embedded_postgres
+              title: Embedded Postgres
+            - name: external_postgres
+              title: External Postgres
+        - name: embedded_postgres_password
+          hidden: true
+          type: password
+          value: "{{repl RandomString 32}}"          
+```
+
+This creates a toggle to allow the user to choose between an embedded or external Postgres instance, and a `hidden` field to generate a unique password for the embedded instance.
+
+As mentioned in the introduction, a full example of the code for this guide can be found in the [kotsapps repository](https://github.com/replicatedhq/kotsapps/tree/master/persistent-datastores).
+
+
+### Validating Config Changes
+
+Even thought the options aren't wired, let's create a new release to validate the configuration screen was modified. Create a release by pushing a commit to your [starter repo copy](../quickstart#automating-your-workflow) or by running `make release` locally. Then head to your kotsadm instance and click "Check for Updates" on the Version History tab to pull the new release:
+ 
+![View Update](/images/guides/kots/view-update.png)
+
+Once the update is deployed, we can head over to the Config tab and review our new toggle. You might also notice that we've removed the Ingress settings to simplify things for this guide:
+
+![Database Config](/images/guides/kots/database-config.png)
+
+Now that we have the configuration screen started, we can proceed to implement the "Embedded Postgres" option.
+
+* * *
+
+## Embedding a Database
+
+To implement the embedded Database option, we'll add a Kubernetes [Statefulset](https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/), and use the [KOTS annotations for optional resources](/vendor/packaging/optional-resources/) to control when it will be included in the application.
+
+### Adding the Secret and StatefulSet
+
+Firse, we'll create a secret to store the root password for our embedded postgres instance. 
+
+```yaml
+# postgres-secret.yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: postgres
```

Comment:
This needs to be called 'postgres-secret', otherwise it will result in an unexpected error.

Reply:
I'm not a fan of including the object type in the object name. It's already a v1/Secret. I'll update the statefulset though, thank you